drivers/SmartThings/matter-switch/src/switch_utils/utils.lua

```diff
@@ -168,60 +168,40 @@ function utils.component_to_endpoint(device, component)
   return utils.find_default_endpoint(device)
 end
 
---- An extension of the library function endpoint_to_component, to support a mapping scheme
---- that includes cluster and attribute id's so that we can use multiple components for a
---- single endpoint.
+--- An extension of the library function endpoint_to_component, used to support a mapping scheme
+--- that optionally includes cluster and attribute ids so that multiple components can be mapped
+--- to a single endpoint.
 ---
 --- @param device any a Matter device object
---- @param opts number|table either is an ep_id or a table { endpoint_id, capability_id }
+--- @param ep_info number|table either an ep_id or a table { endpoint_id, optional(cluster_id), optional(attribute_id) }
```

Comment:
Can we choose a single parameter type and make the the caller provide the additional info? I know lua doesn't have strong parameter typing but I think this is a practice we should try to avoid. If you need to handle additional optional values, they can be passed as optional parameters:
```
function utils.endpoint_to_component(device, ep, cluster_id, attribute_id)
```

Reply:
I don't understand the reasoning for why we cannot add extra arguments - why is backwards compatibility a concern if we are creating a new function that overrides the lua libs? I think we should only use this override when needed, and it shouldn't be a blanket override for all devices in the switch driver, unless they specifically require this advanced mapping, right? 

If this is a function override, I assume we can make it whatever we want. Even using `extend_device` allows us to override the function pointer from the lua libs and just point to this new function, even though I don't neccessarily think using `extend_device` in this case is the way to go.